addons/website_forum/controllers/main.py

```diff
@@ -276,6 +277,37 @@ def forum_post(self, forum, post_type=None, **post):
     def post_create(self, forum, post_parent=None, post_type=None, **post):
         if not request.session.uid:
             return login_redirect()
+
+        if post.get('content', None):
+            validate_karma_content = forum._ckeditor_content_filter(forum, post.get('content', None))
```

Comment:
The check should be done directly in the create / write, in a method that check the content that is posted (called in both methods). This method should raise a KarmaError. Indeed the check is something we intend to have at model-level; the controller has nothing to do with content control.

You should define a json controller that check the content before posting, so no need to define a new template or to redraw the whole page.